cpp/include/cuvs/neighbors/batch_ann.hpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
```

Comment:
I'm trying to keep these specialized "multi-GPU" apis out of the public APIs as much as possible so users don't need to think much about. You can imagine if we have different includes and completley different APIs for single node multi-gpu, multi-node multi-gpu, batched vs non-batched, things will get complicated very quickly. Instead, we've opted for having APIs that can be configured easily (and implicitly where at all possible) so that the single-gpu APIs can function multi-gpu under the proper circumstances. This also means users can change one line of code to use the multi-gpu apis and not have to invoke specialized functions (we are going this way w/ UMAP also).

Reply:
Nevermind this comment- I misunderstood the contents of this PR.